javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,156 @@
+      "Chalk",
+      "Fish fingers",
+      "Socks",
+      "Chocolate",
+      "Toothbrush",
+      "Computer",
+      "Nokia",
+      "Cologne",
+    ];
+
+    let chosenProductName = getRandomItem(productNames);
+    const shouldHavePreWord = getRandomInt(0, 10) > 6;
+
+    if (shouldHavePreWord) {
+      const preWord = preWords[getRandomInt(0, preWords.length - 1)];
+      chosenProductName = `${preWord} ${chosenProductName}`;
+    }
+
+    return chosenProductName;
+  }
+
+  const numberOfAvailableProducts = getRandomInt(0, 30);
+  const availableProducts = Array.apply(
+    null,
+    Array(numberOfAvailableProducts)
+  ).map(() => {
+    const name = getRandomProductname();
+    return {
+      id: `${name}${getRandomInt(0, 100000)}`,
+      name,
+      price: getRandomInt(0, 10000),
+      rating: getRandomInt(1, 10),
+    };
+  });
+
+  return availableProducts;
+};
+
+//
+
+const products = getAvailableProducts();
+console.log(products);
+
+const productList = document.querySelector(".product-list");
+
+function renderProducts(products) {
+  document.getElementById("product-list").innerHTML = "";
+
+  for (let i = 0; i < products.length; i++) {
+    const title = document.createElement("h3");
+    title.classList.add("product-title");
+    title.textContent = products[i].name;
+
+    const price = document.createElement("p");
+    price.classList.add("product-price");
+    price.textContent = `Price: ${products[i].price}`;
+
+    const rating = document.createElement("p");
+    rating.classList.add("product-rating");
+    rating.textContent = `Rating: ${products[i].rating}`;
+
+    productList.appendChild(title);
+    productList.appendChild(price);
+    productList.appendChild(rating);
+  }
+}
+
+renderProducts(products);
+
+const inputForSearch = document.querySelector("#search-item");
+
+inputForSearch.addEventListener("keyup", search);
+
+function search() {
+  const searchInput = document.querySelector("#search-item");
+
+  const searchTerm = searchInput.value;
+  console.log(searchTerm);
+
+  const searchResult = [];
+
+  for (let i = 0; i < products.length; i++) {
+    var name = products[i].name;
+
+    if (name.toUpperCase().includes(searchTerm.toUpperCase())) {
+      searchResult.push(products[i]);
+    }
+  }
+
+  renderProducts(searchResult);
+
+  if (searchResult.length == 0) {
+    const notFound = document.createElement("p");
+    notFound.textContent = "Item not Found !";
+    notFound.classList.add("not-found");
+      productList.appendChild(notFound);
+  }
+}
+
+
+const inputForMaxPrice =document.querySelector("#max-price");
+
+inputForMaxPrice.addEventListener("keyup", setTheMaxPrice);
+
+function setTheMaxPrice (){
+  const maximumPrice= document.querySelector("#max-price");
+  const maximumValue = maximumPrice.value;
+  const maximumPriceResult =[];
+  console.log(maximumValue);
+
+  for (let i = 0; i < products.length; i++) {
+    const price = products[i].price;
+
+   if (price < maximumValue) {
```

Comment:
Would it be more fair , if you use filter here.

Try to rewrite this using filter . same applies for line 120.

Reply:
Hi @shravanipagilla can I get a hint of how to rewrite is with filter?